core/src/main/java/org/testcontainers/containers/wait/strategy/HttpWaitStrategy.java

```diff
@@ -143,6 +157,7 @@ protected void waitUntilReady() {
                 getRateLimiter().doWhenReady(() -> {
                     try {
                         final HttpURLConnection connection = (HttpURLConnection) new URL(uri).openConnection();
+                        connection.setReadTimeout(Math.toIntExact(readTimeout.get(ChronoUnit.MILLIS)));
```

Comment:
IIRC it should be `readTimeout.toMillis()`, otherwise you're only getting the milliseconds part of it

Reply:
Nice, but I have to adjust this tomorrow ;)
I'll leave a comment when I am ready.

Reply:
We plan to release tonight, I can do the review leftovers for you if you wish :)

Reply:
Okay I updated it just now. Please check it and if you need further adjustments feel free to do it yourself if you want ;)